modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/internal/upgrade/v2_5_0/UpgradeFragmentEntryVersion.java

```diff
@@ -0,0 +1,145 @@
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.fragment.internal.upgrade.v2_5_0;
+
+import com.liferay.counter.kernel.service.CounterLocalService;
+import com.liferay.fragment.internal.upgrade.v2_5_0.util.FragmentEntryVersionTable;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.sql.Timestamp;
+
+/**
+ * @author Rubén Pulido
+ */
+public class UpgradeFragmentEntryVersion extends UpgradeProcess {
+
+	public UpgradeFragmentEntryVersion(
+		CounterLocalService counterLocalService) {
+
+		_counterLocalService = counterLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasTable(FragmentEntryVersionTable.TABLE_NAME)) {
+			runSQL(FragmentEntryVersionTable.TABLE_SQL_CREATE);
+		}
+
+		insertIntoFragmentEntryVersion();
+	}
+
+	protected void insertIntoFragmentEntryVersion() throws Exception {
+		StringBundler sb1 = new StringBundler(11);
+
+		sb1.append("select uuid_, fragmentEntryId, groupId, companyId, ");
+		sb1.append("userId, userName, createDate, modifiedDate, ");
+		sb1.append("fragmentCollectionId, fragmentEntryKey, name, css, html, ");
+		sb1.append("js, cacheable, configuration, previewFileEntryId, ");
+		sb1.append("readOnly, type_, lastPublishDate, status, ");
+		sb1.append("statusByUserId, statusByUserName, statusDate from ");
+		sb1.append("FragmentEntry where status = ");
+		sb1.append(WorkflowConstants.STATUS_APPROVED);
+		sb1.append(" and FragmentEntry.fragmentEntryId not in (select ");
+		sb1.append("FragmentEntryVersion.fragmentEntryId from ");
+		sb1.append("FragmentEntryVersion)");
+
+		StringBundler sb2 = new StringBundler(9);
+
+		sb2.append("insert into FragmentEntryVersion(fragmentEntryVersionId, ");
+		sb2.append("version, uuid_, fragmentEntryId, groupId, companyId, ");
+		sb2.append("userId, userName, createDate, modifiedDate, ");
+		sb2.append("fragmentCollectionId, fragmentEntryKey, name, css, html, ");
+		sb2.append("js, cacheable, configuration, previewFileEntryId, ");
+		sb2.append("readOnly, type_, lastPublishDate, status, ");
+		sb2.append("statusByUserId, statusByUserName, statusDate) values(?, ");
+		sb2.append("?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ");
+		sb2.append("?, ?, ?, ?, ?, ?)");
+
+		try (Statement s = connection.createStatement();
+			PreparedStatement ps =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection, sb2.toString())) {
```

Comment:
Since you're copying every column you can simplify this with a update from select query.  It'll be faster and much more readable.

Reply:
You can hack the fragmentEntryVersionId by just setting it to the same value as the fragmentEntryId and insert a new FragmentEntryVersion.class.getName() counter at the correct value.